src/vs/platform/environment/node/stdin.ts

```diff
@@ -51,15 +48,21 @@ export async function readFromStdin(targetPath: string, verbose: boolean): Promi
 	}
 
 	// Pipe into tmp file using terminals encoding
+	// Use a `Queue` to be able to use `appendFile`
+	// which helps file watchers to be aware of the
+	// changes because each append closes the underlying
+	// file descriptor.
+	// (https://github.com/microsoft/vscode/issues/148952)
 	const decoder = iconv.getDecoder(encoding);
-	process.stdin.on('data', chunk => stdinFileStream.write(decoder.write(chunk)));
+	const appendFileQueue = new Queue();
+	process.stdin.on('data', chunk => {
+		const chunkStr = decoder.write(chunk);
+		appendFileQueue.queue(() => Promises.appendFile(targetPath, chunkStr));
```

Comment:
At some point we should start removing all the promisify stubs in here and just use `fs.promises`.

Reply:
@joaomoreno we cannot, `graceful-fs` does not support patching `fs.promises`.

Reply:
https://github.com/microsoft/vscode/blob/32d40cf44e893e87ac33ac4f08de1e5f7fe077fc/src/vs/base/node/pfs.ts#L616-L631

Also, there are subtle differences it seems in how the methods are actually implemented for some...